File: 1461-check-if-a-string-contains-all-binary-codes-of-size-k/1461-check-if-a-string-contains-all-binary-codes-of-size-k.cpp
class Solution {
public:
    bool hasAllCodes(string s, int k) {
        unordered_set<string> st;

        for (int i = 0; i + k <= s.size(); i++) {
            st.insert(s.substr(i, k));
        }

        return st.size() == (int)pow(2, k);
    }
};

File: 1461-check-if-a-string-contains-all-binary-codes-of-size-k/1461-check-if-a-string-contains-all-binary-codes-of-size-k.java
import java.util.*;
class Solution {
    public boolean hasAllCodes(String s, int k) {
        HashSet<String> st = new HashSet<>();
        for (int i = 0; i + k <= s.length(); i++) {
            st.add(s.substring(i, i + k));
        }
        return st.size() == (int)Math.pow(2, k);
    }
}